src/types/cursor/xcursor_manager.rs

```diff
@@ -0,0 +1,38 @@
+use wlroots_sys::{wlr_xcursor_manager, wlr_xcursor_manager_create, wlr_xcursor_manager_load, wlr_xcursor_manager_set_cursor_image};
+use types::Cursor;
+use std::ptr;
+use utils::safe_as_cstring;
+
+#[derive(Debug)]
+pub struct XCursorManager {
+    manager: *mut wlr_xcursor_manager
+}
+
+impl XCursorManager {
+    pub fn create<T: Into<Option<String>>>(name: T, size: u32) -> Option<Self> {
+        unsafe {
+            let name_str = name.into().map(safe_as_cstring);
+            let name_ptr = name_str.map(|s| s.as_ptr()).unwrap_or(ptr::null_mut());
+            let manager = wlr_xcursor_manager_create(name_ptr, size);
+            if manager.is_null() {
+                None
+            } else {
+                Some(XCursorManager { manager })
+            }
+        }
+    }
+
+    /// returns 0 if the scaled theme was successfully loaded and 1 otherwise
+    pub fn load(&self, scale: f32) -> i32 {
+        unsafe {
+            wlr_xcursor_manager_load(self.manager, scale)
+        }
+    }
+
+    pub fn set_cursor_image<T: Into<String>>(&self, name: T, cursor: &Cursor) {
```

Comment:
Why is this `Into<String>` instead of simply `String` for name?

Should be `&mut cursor` since we mutating the cursor (and we should reflect that in the API).

Reply:
Wouldn't an `AsRef<str>` work?

Reply:
When I read it, I thought it would let me pass anything that could be converted into a string as a parameter, but now I don't think it does that.

I have no idea what I'm doing.

Reply:
> Wouldn't an AsRef<str> work?

There is a case to be made for a reference since this will normally be passed as a literal.

Reply:
> When I read it, I thought it would let me pass anything that could be converted into a string as a parameter, but now I don't think it does that.

It does! I probably should have been more clear why I think this should be `String`. Right now this is hiding an allocation from user code, when we should be upfront about the need for an allocation. So we should use `String` as the input and force the user of the code to put `"some literal string".into()` (or `to_string`, or any of the other ways to convert a `&str` to a `String` in Rust) so that they are obviously opting into an allocation.